test/ng/compileSpec.js

```diff
@@ -1201,41 +1201,50 @@ describe('$compile', function() {
           });
         });
 
-        it('should fail if replacing and template doesn\'t have a single root element', function() {
-          module(function() {
-            directive('noRootElem', function() {
-              return {
-                replace: true,
-                template: 'dada'
-              };
-            });
-            directive('multiRootElem', function() {
-              return {
-                replace: true,
-                template: '<div></div><div></div>'
-              };
-            });
-            directive('singleRootWithWhiteSpace', function() {
+        describe('replace and not exactly one root element', function() {
+          var templateVar;
+
+          beforeEach(module(function() {
+            directive('template', function() {
               return {
                 replace: true,
-                template: '  <div></div> \n'
+                template: function() {
+                  return templateVar;
+                }
               };
             });
-          });
+          }));
 
-          inject(function($compile) {
-            expect(function() {
-              $compile('<p no-root-elem></p>');
-            }).toThrowMinErr('$compile', 'tplrt', 'Template for directive \'noRootElem\' must have exactly one root element. ');
+          they('should throw if: $prop',
+            {
+              'no root element': 'dada',
+              'multiple root elements': '<div></div><div></div>'
+            }, function($prop) {
```

Comment:
Calling this `$prop` too is confusing (since it is not the same `$prop` as in the description).
Maybe `template` or something.